compile_protobuf.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# Standard Library
+import glob
+import logging
+import os
+import shutil
+import sys
+import tempfile
+import urllib
+import urllib.request
+from subprocess import check_call
+from zipfile import ZipFile
+
+
+def script_name() -> str:
+    """:returns: script name with leading paths removed"""
+    return os.path.split(sys.argv[0])[1]
+
+
+def config_logging():
+    import time
+
+    logging.getLogger().setLevel(logging.INFO)
+    logging.basicConfig(format="{}: %(asctime)sZ %(levelname)s %(message)s".format(script_name()))
+    logging.Formatter.converter = time.gmtime
+
+
+def _protoc_bundle():
+    """
+    Returns an archive with the binary protoc distro for the platform
+    """
+    import platform
+
+    system = platform.system()
+    machine = platform.machine()
+    if system == "Darwin":
+        archive_url = "https://github.com/protocolbuffers/protobuf/releases/download/v3.11.4/protoc-3.11.4-osx-x86_64.zip"
+        logging.info("Downloading protoc for Darwin: %s", archive_url)
+    elif system == "Linux":
+        archive_url = f"https://github.com/protocolbuffers/protobuf/releases/download/v3.11.4/protoc-3.11.4-linux-{machine}.zip"
+        logging.info("Downloading protoc for Linux: %s", archive_url)
+    else:
+        archive_url = None
+    return archive_url
+
+
+def get_protoc():
+    """make sure protoc is available, otherwise download it and return a tuple with the protoc
+    binary and a temporary dir if it needed to be downloaded"""
+    if not shutil.which("protoc"):
```

Comment:
Restructure this condition to return early if protoc is already available.

```
if not shutil.which("protoc"):
    return (shutil.which("protoc"), None)

archive_url = ...
...
    
```

This makes the function easier to read.